src/System.Memory/src/System/SpanHelpers.T.cs

```diff
@@ -119,6 +119,109 @@ public static unsafe int IndexOf<T>(ref T searchSpace, T value, int length)
             return (int)(byte*)(index + 7);
         }
 
+        public static int LastIndexOf<T>(ref T searchSpace, int searchSpaceLength, ref T value, int valueLength)
+    where T : IEquatable<T>
+        {
+            Debug.Assert(searchSpaceLength >= 0);
+            Debug.Assert(valueLength >= 0);
+
+            if (valueLength == 0)
+                return 0;  // A zero-length sequence is always treated as "found" at the start of the search space.
+
+            T valueHead = value;
+            ref T valueTail = ref Unsafe.Add(ref value, 1);
+            int valueTailLength = valueLength - 1;
+
+            int index = 0;
+            for (;;)
```

Comment:
So if span is "abaaaaaaaaaaa..." and I search for "ab", it will do lots of loops. Maybe there is nothing we can do.

Reply:
The same would apply to something like "...aaaaaaaaab" and you search for "ab" when you call IndexOf (which is where I borrowed the template for LastIndexOf). There may be a different approaches that balance things better but I haven't thought too deeply about it yet since it would require a bigger change (to all our *IndexOf* APIs).